language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/components/TemplateA.java

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.rxjava.components;
+
+import com.daml.ledger.javaapi.data.*;
+import java.util.List;
+
+public class TemplateA extends Template {
```

Comment:
If the intent is to make sure that this integrates with codegen, then codegen should be used to make sure we're not breaking the utility you added. You can use the approach used by the Java codegen integration tests:
- write the piece of DAML you want to interact with
- run codegen on it
- use the generated code for your tests
You can read the Bazel code we wrote to encode this process [here](https://github.com/digital-asset/daml/blob/master/language-support/java/codegen/BUILD.bazel#L163-L240).
Note that for what you want to achieve, probably using just the latest DAML-LF version is enough, so the complexity of the pipeline should be way easier that this. I'm 100% available for help if you need any.

Reply:
Agreed, I'd rather chose a real code generation, I'm trying to accomplish that.

Reply:
Hopefully it's done.